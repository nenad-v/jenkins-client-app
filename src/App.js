import React from 'react';
import {useEffect, useState} from 'react';
import axios from 'axios';

const api = 'http://localhost:5000/api';

const App = () => {
    const [posts, setPosts] = useState([]);

    useEffect(() => {
        axios.get(api + '/posts')
            .then(res => {
                setPosts(res.data)
            })
    }, [])

    return (
        <div className="app">
            {
                posts.map(post => (
                    <div key={post.id}>
                        <h4>{post.title}</h4>
                    </div>
                ))
            }
        </div>
    )
}

export default App;