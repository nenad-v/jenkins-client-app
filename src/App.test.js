import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('When I render App component', () => {
    it('should render correctly', () => {
        const component = shallow(<App />);
        expect(component.html()).toMatchSnapshot();
        expect(component.exists('.app')).toEqual(true)
    })
})