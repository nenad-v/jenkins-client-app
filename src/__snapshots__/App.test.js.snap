// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I render App component should render correctly 1`] = `"<div class=\\"app\\"></div>"`;
